include/ppx/geometry.h

```diff
@@ -152,47 +152,46 @@ class Geometry
 
         BufferType  GetType() const { return mType; }
         uint32_t    GetElementSize() const { return mElementSize; }
-        uint32_t    GetSize() const { return CountU32(mData); }
-        void        SetSize(uint32_t size) { mData.resize(size); }
+        uint32_t    GetSize() const { return static_cast<uint32_t>(mUsedSize); }
         char*       GetData() { return DataPtr(mData); }
         const char* GetData() const { return DataPtr(mData); }
         uint32_t    GetElementCount() const;
 
+        // Changes the buffer size to `size`.
+        // If this operations grows the vector, this class considers all the elements
+        // to be initialized. It is the caller responsibility to initialize those elements.
+        void SetSize(uint32_t size)
+        {
+            mUsedSize = size;
```

Comment:
Wondering if we should emit a warning if size is less than the current size? Or perhaps if the caller is going this way then we really trust them to know what they are doing?

Reply:
I do see why we would want that, but it's also a valid behavior to shrink an array.
I see this buffer more as a small wrapper over vector, which allows the same resize & append behaviors as we do without warning.